### ISSUE TITLE
StrMatch plus_coercions broken in >= 1.015_000

### ISSUE THREAD

**Issue by sbuggles** [tobyink/p5-type-tiny#96]
Hi, I hope this report is helpful. I'm unsure if I'm using `plus_coercions` correctly, but my code worked until I upgraded to the latest Type::Tiny.

Here's a one-file example of what's happening:

```perl
#!/usr/bin/env perl

use v5.12;

package MyBookList {

    use Readonly;

    use Moo;
    use Types::Standard qw( StrMatch InstanceOf );
    use Type::Params    qw( compile wrap_subs Invocant );

    Readonly(
        my $UUID_RE => qr{
        ^
        [0-9a-fA-F]{8}-
        [0-9a-fA-F]{4}-
        [0-9a-fA-F]{4}-
        [0-9a-fA-F]{4}-
        [0-9a-fA-F]{12}
        $
    }sxm
    );

    wrap_subs remove_book => compile( Invocant,
        StrMatch( [$UUID_RE] )
          ->plus_coercions( InstanceOf ['MyBook'], sub { shift->uuid } ) );

    sub remove_book {
        my $self = shift;
        my $uuid = shift;

        say "Attempting to remove $uuid";

        return $self;
    }

}

package MyBook {
    use Moo;

    has name => ( is => 'rw' );
    has uuid => ( is => 'rw' );
}

use MyBook;

my $book1 = MyBook->new(
    title => 'Doon',
    uuid  => '12345678-9999-9999-9999-123456789012'
);

my $m = MyBookList->new();
$m->remove_book('12345678-1234-1234-1234-123456789012');    # Remove by UUID
$m->remove_book($book1);    # Remove by MyBook object
```
with Tiny::Type version `1.014000` the code parses cleanly and runs correctly, 

```console
$ perl -MType::Tiny -e 'print $Type::Tiny::VERSION ."\n";'
1.014000
$ ./onefile.pl
Attempting to remove 12345678-1234-1234-1234-123456789012
Attempting to remove 12345678-9999-9999-9999-123456789012
```

but starting from `1.015_000` I get variations of this

```console
$ perl -MType::Tiny -e 'print $Type::Tiny::VERSION ."\n";'
1.016008
$ ./onefile.pl 
Bareword found where operator expected at parameter validation for __ANON__::__ANON__ line 14, near "9a"
	(Missing operator before a?)
Bareword found where operator expected at parameter validation for __ANON__::__ANON__ line 15, near "9a"
	(Missing operator before a?)
Bareword found where operator expected at parameter validation for __ANON__::__ANON__ line 16, near "9a"
	(Missing operator before a?)
Bareword found where operator expected at parameter validation for __ANON__::__ANON__ line 17, near "9a"
	(Missing operator before a?)
Bareword found where operator expected at parameter validation for __ANON__::__ANON__ line 18, near "9a"
	(Missing operator before a?)
Scalar found where operator expected at parameter validation for __ANON__::__ANON__ line 20, near "}
	        $
	    )"
	(Missing operator before 	    )?)
Scalar found where operator expected at parameter validation for __ANON__::__ANON__ line 21, near ")
	$tmp"
	(Missing operator before $tmp?)
Failed to compile source because: syntax error at parameter validation for __ANON__::__ANON__ line 13, near ";

	# Parameter $_[1] (type: StrMatch[(?^umsx:
	        ^"
syntax error at parameter validation for __ANON__::__ANON__ line 15, near "9a"
syntax error at parameter validation for __ANON__::__ANON__ line 16, near "9a"
Global symbol "$tmp" requires explicit package name (did you forget to declare "my $tmp"?) at parameter validation for __ANON__::__ANON__ line 21.
Global symbol "$coercion_for_arg_1" requires explicit package name (did you forget to declare "my $coercion_for_arg_1"?) at parameter validation for __ANON__::__ANON__ line 21.
Global symbol "$tmp" requires explicit package name (did you forget to declare "my $tmp"?) at parameter validation for __ANON__::__ANON__ line 22.
Global symbol "$tmp" requires explicit package name (did you forget to declare "my $tmp"?) at parameter validation for __ANON__::__ANON__ line 22.
Global symbol "$tmp" requires explicit package name (did you forget to declare "my $tmp"?) at parameter validation for __ANON__::__ANON__ line 23.
Global symbol "@out" requires explicit package name (did you forget to declare "my @out"?) at parameter validation for __ANON__::__ANON__ line 24.
Global symbol "$tmp" requires explicit package name (did you forget to declare "my $tmp"?) at parameter validation for __ANON__::__ANON__ line 24.
Global symbol "@out" requires explicit package name (did you forget to declare "my @out"?) at parameter validation for __ANON__::__ANON__ line 26.
syntax error at parameter validation for __ANON__::__ANON__ line 27, near ";
}"
parameter validation for __ANON__::__ANON__ has too many errors. at /home/sbg/perl5/perlbrew/perls/TypeTinyTest/lib/site_perl/5.36.0/Eval/TypeTiny/CodeAccumulator.pm line 113.
```

I can work around it. For now, I'm locking my container to `1.014000`.

**Comment by tobyink** [comment]
Ah, it's because the regexp contains new line characters.

This results in the type's display_name having new line characters. Type::Params attempts to put that name in a comment, but Perl comments end at the end of a line.

I'll fix this.

**Comment by tobyink** [comment]
Released 1.016009 with a fix.
